services/circleci/circleci.service.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+
+const circleSchema = Joi.array()
+  .items(Joi.object({ status: Joi.string().required() }))
+  .min(1)
+  .max(1)
+  .required()
+
+module.exports = class CircleCi extends BaseJsonService {
+  async fetch({ token, vcsType, userRepo, branch }) {
+    let url = `https://circleci.com/api/v1.1/project/${vcsType}/${userRepo}`
+    if (branch != null) {
+      url += `/tree/${branch}`
+    }
+    const query = { filter: 'completed', limit: 1 }
+    if (token) {
+      query['circle-token'] = token
+    }
+    return this._requestJson({
+      url,
+      schema: circleSchema,
+      options: { qs: query },
+      errorMessages: { 404: 'project not found' },
+    })
+  }
+
+  static render({ status }) {
+    if (['success', 'fixed'].includes(status)) {
+      return { message: 'passing', color: 'brightgreen' }
+    } else if (status === 'failed') {
+      return { message: 'failed', color: 'red' }
+    } else if (['no_tests', 'scheduled', 'not_run'].includes(status)) {
+      return { message: status.replace('_', ' '), color: 'yellow' }
+    } else {
+      return { message: status.replace('_', ' '), color: 'lightgrey' }
+    }
+  }
+
+  async handle({ token, vcsType, userRepo, branch }) {
+    const json = await this.fetch({ token, vcsType, userRepo, branch })
+    return this.constructor.render({ status: json[0].status })
+  }
+
+  // Metadata
+  static get defaultBadgeData() {
+    return { label: 'build' }
```

Comment:
Do you think it would be worth leaving this out as it defaults to the `category` already?

Reply:
Having chewed this over, I think it is useful to keep labels explicit.

It is helpful that we can fall back to category to ensure we do have a label, but if you think about the re-organisation work going on in https://github.com/badges/shields/issues/1905#issue-349816846 I wouldn't want to be in a situation where we move badges around and all the labels change.

For example, if we renamed the 'Build' category to 'Continuous Integration' (not saying we should do this - just an example), we would probably still want this label to be 'build'. Lets not create a situation where doing something like that creates a domino effect causing unpredictable changes elsewhere.